Add dual panda environment

cambel opened the pull request:
Adding a dual-arm environment using the `Panda` robot for testing features regarding multi-arms.
URDF models were taken from [franka_description](https://github.com/frankaemika/franka_ros/tree/noetic-devel/franka_description/robots)'s `noetic-devel`. This is a self-contained package, with no reference to any external packages.

See MoveIt's PR [3119](https://github.com/ros-planning/moveit/pull/3119) and [2810](https://github.com/ros-planning/moveit/pull/2810).

simonschmeisser commented:
I was slightly confused why this PR includes the panda_arm.urdf.xacro because I expected this to be present already. But since this repo only contains the "compiled" urdf so far, it makes sense to include it here. Maybe in a follow up PR this could be extended to remove the URDF and use the xacro directly everywhere?

This fails with some line-ending errors by pre-commit, I recommend installing it to fix stuff like this automatically

cambel commented:
@simonschmeisser Thank you for checking my PR. Yes, I had to add the xacro to make it easier to recycle the urdf. 
Also, I think is a good idea to replace the compiled urdf with the xacro version.

I added pre-commit and pushed the changes.

rhaschke commented on panda_description/urdf/dual_panda.urdf.xacro:
```diff
@@ -0,0 +1,36 @@
+  <xacro:arg name="arm_id_1" default="panda_1" />
+  <xacro:arg name="arm_id_2" default="panda_2" />
+
+  <xacro:include filename="$(find moveit_resources_panda_description)/urdf/panda_arm.xacro"/>
+  <xacro:include filename="$(find moveit_resources_panda_description)/urdf/hand.xacro"/>
```
Here you include xacro files from `moveit_resources_panda_description`. Thus, as pointed out before by Simon, there is no need to include those xacro files in this package again.
However, you need to add a run-time dependency to `moveit_resources_panda_description`.

rhaschke replied:
Sorry, just have seen that you added those xacro files to panda_description.

rhaschke reviewed (commented):
Please get rid of extra xacro files. Otherwise looks good.

rhaschke reviewed (commented):
@cambel, please could you explain from which version of `franka_description` you copied the xacro files from? This should be definitely stated within the files.
I'm asking because, Franka screwed up the urdf description with release 0.8.0. They merged a fix (https://github.com/frankaemika/franka_ros/pull/199), but didn't release a new version yet. Thus, I'm afraid your version is still broken.

cambel commented:
Hi @rhaschke,

I used the 0.7.1 version of the `franka_description` package. I remember trying the newest one but having trouble with it, so I just used an older version. I added a comment describing the version of the files.
I can update the files once the fixed version is released.

I also added the runtime dependency to `moveit_resources_panda_description`.

simonschmeisser commented on dual_panda_moveit_config/launch/gazebo.launch:
```diff
@@ -0,0 +1,32 @@
+  </include>
+
+  <!-- send robot urdf to param server -->
+  <param name="robot_description" textfile="$(find moveit_resources_dual_panda_moveit_config)/config/gazebo_panda.urdf" />
```
All other places generate the urdf from xacro on the fly. Can this be used here as well? (I'm using gazebo ...)

simonschmeisser replied:
@rhaschke @v4hn Is it customary to store and commit the compiled and extended URDF for gazebo or would you rather copy those additions to the xacro? A headless tool that could be piped through to add those modifications does not exist right?

rhaschke replied:
We could actually drop the Gazebo-compatible URDF altogether. We don't have it for panda_description as well.
In any case, if Franka's panda_description will be released, we should consider migrating to that one. Unfortunately, the current approach uses different kinematics?, visuals, and geometries for panda.urdf (single config) than for dual_panda.urdf.xacro.

> A headless tool that could be piped through to add those modifications does not exist right?

What modifications do you refer to? Adding Gazebo-specific stuff? For those, we only have MSA for now.
